### PR TITLE
Force node restart during initialization on desktop

### LINKED ISSUE #5444: App doesn't start after crash
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to restart app after crash and continue work.

### Description

*Type*: Bug

*Summary*: when application is crashing, and you try to restart it (for crash example with logs see https://github.com/status-im/status-react/issues/5012) when you open app nothing happens. To continue work you should manually kill `ubuntu-server` process and then restart the app.

#### Expected behavior
app is working when you restart it after crash

#### Actual behavior
nothing happens until you kill `ubuntu-server` process 

### Reproduction

- when you have a crash, restart the app.

### Additional Information
* Status version: nightly 09.08.2017
* Operating System: Mac OSx

### LINKED ISSUE #6061: Reloading JS in Desktop freezes the app
This is caused by an attempt to start a Geth node that is already running. A solution would be to always stop and start node during desktop app startup.

### PR DESCRIPTION
**Pull request by vitvly**
Fixes #6061 

### QA notes:
This can be tested in dev mode. Steps:
  - Launch app by following steps outlined in https://docs.status.im/docs/build_desktop.html
  - Press `Cmd+R` once app is running. There, click `Reload`
  - App should reload with no issues. Previously, it would hang on `Signing in...` step

### CODE REVIEW

**Review by rasom — APPROVED** [review]

File: src/status_im/node/models.cljs [hunk 1]
```diff
@@ -49,6 +50,6 @@
 
 (defn initialize
   [address {{:keys [status-node-started?] :as db} :db :as cofx}]
-  (if (not status-node-started?)
-    (start address cofx)
-    (restart)))
+  (if (or status-node-started? platform/desktop?)
```
Comment (vkjr) [inline src/status_im/node/models.cljs]: @siphiuel, don't we have `status-node-started?` correctly set when running desktop build?

Comment (vitvly) [reply src/status_im/node/models.cljs]: Good question! Just checked - we have it set during the first run, but if we e.g. `Reload JS`, it becomes `nil`.

**Review by maxhora — APPROVED** [review]

**Comment by churik** [comment]
@siphiuel IOS and android build in this PR were failed:
Link to [build](https://ci.status.im/job/status-react/job/pull%20requests/view/change-requests/job/PR-5784/).
Can you please check?

**Comment by churik** [comment]
@siphiuel 

if I did "Force quit" `status` process in Activity Monitor, `ubuntu-server` remains open, and then I start Status app again - nothing happens, as it was before.
I can only see icon in "Dock" and `status` process.
Log: [StatusIm.log](https://github.com/status-im/status-react/files/2396803/StatusIm.log)

**Review by vkjr — APPROVED** [review]

**Review by vkjr — APPROVED** [review]

**Comment by rasom** [comment]
@siphiuel @churik any updates on this?

**Comment by vitvly** [comment]
@churik @rasom I've rebased it against latest develop. This is a dev-only change - as stated in referenced issue #6061, it fixes the problem of app hanging when clicking `Reload` inside RN menu of the app. Not sure if this requires manual testing. Maybe we can merge it as is. Please advise.

**Comment by churik** [comment]
@siphiuel 
if it is OK https://github.com/status-im/status-react/pull/5784#issuecomment-422763628 according to your changes, probably I misunderstood smth.
So feel free to comment.

**Comment by vitvly** [comment]
@churik, yes the above issue is not fixed by this PR. It will not affect non-dev builds.

**Comment by churik** [comment]
@siphiuel 
after building it locally got "Failed to load sorce code"
1) Log from `react-native run-desktop`:
```
JS server already running.
Starting the app...
hostname: illegal option -- I
usage: hostname [-fs] [name-of-host]
"ViewManager:" Created view:  ReactActivityIndicator_QMLTYPE_4(0x7fa15bfdbc70, parent=0x0, geometry=0,0 12x12) . Source QML file:  "qrc:/qml/ReactActivityIndicator.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactButton_QMLTYPE_6(0x7fa15bfe80e0, parent=0x0, geometry=0,0 16x12) . Source QML file:  "qrc:/qml/ReactButton.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactImage_QMLTYPE_17(0x7fa161c02cd0, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactImage.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactModal_QMLTYPE_18(0x7fa15d0903c0, name="modalRoot", parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactModal.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactNavigator_QMLTYPE_19(0x7fa161c27030, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactNavigator.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactPicker_QMLTYPE_21(0x7fa161c2e6f0, parent=0x0, geometry=0,0 60x200) . Source QML file:  "qrc:/qml/ReactPicker.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactRawText_QMLTYPE_22(0x7fa15bd9ddf0, parent=0x0, geometry=0,0 0x16) . Source QML file:  "qrc:/qml/ReactRawText.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactScrollView_QMLTYPE_23(0x7fa161c34300, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactScrollView.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactSlider_QMLTYPE_24(0x7fa15bda54d0, parent=0x0, geometry=0,0 100x40) . Source QML file:  "qrc:/qml/ReactSlider.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactSwitch_QMLTYPE_26(0x7fa15d09b3d0, parent=0x0, geometry=0,0 74x40) . Source QML file:  "qrc:/qml/ReactSwitch.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactText_QMLTYPE_28(0x7fa15d0afca0, parent=0x0, geometry=0,0 0x16) . Source QML file:  "qrc:/qml/ReactText.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactTextInput_QMLTYPE_29(0x7fa15bee9f20, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactTextInput.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactView_QMLTYPE_31(0x7fa161c614c0, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactView.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactView_QMLTYPE_31(0x7fa161c686b0, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactView.qml" . Props keys:  ()
WebEngineContext used before QtWebEngine::initialize() or OpenGL context creation failed.
"ViewManager:" Created view:  ReactWebView_QMLTYPE_32(0x7fa161c712a0, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactWebView.qml" . Props keys:  ()
QVariantMap (anonymous namespace)::makeReactTouchEvent(QQuickItem *, QMouseEvent *) target was not a reactItem
QVariantMap (anonymous namespace)::makeReactTouchEvent(QQuickItem *, QMouseEvent *) target was not a reactItem
auto SourceCode::loadSource(QNetworkAccessManager *)::(anonymous class)::operator()() const : Error while loading source "Error transferring http://localhost:8081/index.desktop.bundle?platform=desktop&dev=true - server replied: Internal Server Error"
"ViewManager:" Created view:  ReactActivityIndicator_QMLTYPE_4(0x7fa161d27890, parent=0x0, geometry=0,0 12x12) . Source QML file:  "qrc:/qml/ReactActivityIndicator.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactButton_QMLTYPE_6(0x7fa161d309c0, parent=0x0, geometry=0,0 16x12) . Source QML file:  "qrc:/qml/ReactButton.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactImage_QMLTYPE_17(0x7fa161d35320, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactImage.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactModal_QMLTYPE_18(0x7fa161d4e250, name="modalRoot", parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactModal.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactNavigator_QMLTYPE_19(0x7fa161d56580, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactNavigator.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactPicker_QMLTYPE_21(0x7fa161d59730, parent=0x0, geometry=0,0 60x200) . Source QML file:  "qrc:/qml/ReactPicker.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactRawText_QMLTYPE_22(0x7fa161d625d0, parent=0x0, geometry=0,0 0x16) . Source QML file:  "qrc:/qml/ReactRawText.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactScrollView_QMLTYPE_23(0x7fa161d65230, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactScrollView.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactSlider_QMLTYPE_24(0x7fa161d6c5b0, parent=0x0, geometry=0,0 100x40) . Source QML file:  "qrc:/qml/ReactSlider.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactSwitch_QMLTYPE_26(0x7fa161d74980, parent=0x0, geometry=0,0 74x40) . Source QML file:  "qrc:/qml/ReactSwitch.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactText_QMLTYPE_28(0x7fa161cae540, parent=0x0, geometry=0,0 0x16) . Source QML file:  "qrc:/qml/ReactText.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactTextInput_QMLTYPE_29(0x7fa161c094d0, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactTextInput.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactView_QMLTYPE_31(0x7fa161cb6a70, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactView.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactView_QMLTYPE_31(0x7fa161cbe710, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactView.qml" . Props keys:  ()
"ViewManager:" Created view:  ReactWebView_QMLTYPE_32(0x7fa161cc6510, parent=0x0, geometry=0,0 0x0) . Source QML file:  "qrc:/qml/ReactWebView.qml" . Props keys:  ()
auto SourceCode::loadSource(QNetworkAccessManager *)::(anonymous class)::operator()() const : Error while loading source "Error transferring http://localhost:8081/index.desktop.bundle?platform=desktop&dev=true - server replied: Internal Server Error"
```
2) Log from `lein figwheel-repl desktop`:
```
Successfully compiled build :desktop to "target/desktop/app.js" in 37.974 seconds.
Figwheel: Starting nREPL server on port: 7888
WARNING: unable to load "cemerick.piggieback/wrap-cljs-repl" middleware
re-frisk server has been started at http://localhost:4567
Launching ClojureScript REPL for build: desktop
Figwheel Controls:
          (stop-autobuild)                ;; stops Figwheel autobuilder
          (start-autobuild id ...)        ;; starts autobuilder focused on optional ids
          (switch-to-build id ...)        ;; switches autobuilder to different build
          (reset-autobuild)               ;; stops, cleans, and starts autobuilder
          (reload-config)                 ;; reloads build config and resets autobuild
          (build-once id ...)             ;; builds source one time
          (clean-builds id ..)            ;; deletes compiled cljs target files
          (print-config id ...)           ;; prints out build configurations
          (fig-status)                    ;; displays current state of system
          (figwheel.client/set-autoload false)    ;; will turn autoloading off
          (figwheel.client/set-repl-pprint false) ;; will turn pretty printing off
  Switch REPL build focus:
          :cljs/quit                      ;; allows you to switch REPL to another build
    Docs: (doc function-name-here)
    Exit: :cljs/quit
 Results: Stored in vars *1, *2, *3, *e holds last exception object
Prompt will show when Figwheel connects to your application
```
3) Log from `npm start`:
```
 BUNDLE  [desktop, dev] ./index.desktop.js ░░░░░░░░░░░░░░░░ 0.0% (0/1), failed.

error: bundling failed: Error: Unable to resolve module `./resources/images/assets/ethereum.png` from `/Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react/index.desktop.js`: The module `./resources/images/assets/ethereum.png` could not be found from `/Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react/index.desktop.js`. Indeed, none of these files exist:

  * `ethereum.png`
  * `/Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react/resources/images/assets/ethereum.png/index(.native||.desktop.js|.native.js|.js|.desktop.json|.native.json|.json)`
    at ModuleResolver.resolveDependency (/Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:167:697)
    at ResolutionRequest.resolveDependency (/Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react/node_modules/metro/src/node-haste/DependencyGraph/ResolutionRequest.js:80:16)
    at DependencyGraph.resolveDependency (/Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react/node_modules/metro/src/node-haste/DependencyGraph.js:237:485)
    at Object.resolve (/Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react/node_modules/metro/src/lib/transformHelpers.js:116:25)
    at dependencies.map.result (/Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react/node_modules/metro/src/DeltaBundler/traverseDependencies.js:298:29)
    at Array.map (<anonymous>)
    at resolveDependencies (/Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react/node_modules/metro/src/DeltaBundler/traverseDependencies.js:294:16)
    at /Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react/node_modules/metro/src/DeltaBundler/traverseDependencies.js:159:33
    at Generator.next (<anonymous>)
    at step (/Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react/node_modules/metro/src/DeltaBundler/traverseDependencies.js:239:307)
```

**Comment by vitvly** [comment]
@churik can you please running it locally again?

**Comment by churik** [comment]
@siphiuel I can't. When use `npm start` got:
```
MacBook-Pro:status-react curikovatm$ npm start
npm ERR! missing script: start

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/curikovatm/.npm/_logs/2018-10-25T15_10_45_324Z-debug.log
```
Log:
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'start' ]
2 info using npm@5.5.1
3 info using node@v8.11.2
4 verbose stack Error: missing script: start
4 verbose stack     at run (/usr/local/lib/node_modules/npm/lib/run-script.js:151:19)
4 verbose stack     at /usr/local/lib/node_modules/npm/lib/run-script.js:61:5
4 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/read-package-json/read-json.js:115:5
4 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/read-package-json/read-json.js:402:5
4 verbose stack     at handleExists (/usr/local/lib/node_modules/npm/node_modules/read-package-json/read-json.js:366:20)
4 verbose stack     at FSReqWrap.cb [as oncomplete] (fs.js:312:19)
5 verbose cwd /Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react
6 verbose Darwin 17.7.0
7 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
8 verbose node v8.11.2
9 verbose npm  v5.5.1
10 error missing script: start
11 verbose exit [ 1, true ]
```
What else can I do to start it?

**Comment by vitvly** [comment]
Weird. Maybe the reason is there were some updates to the way app is started now, and the docs website is not updated yet. Sorry I didn't mention it earlier - I did a rebase today so the startup procedure has changed. Please do the below after cloning the branch into status-react:

1. make clean
2. make prepare-desktop
3. Now in 4 separate tabs:
    - npm start
    - node ./ubuntu-server.js
    - make watch-desktop (wait to finish)
    - react-native run-desktop

**Comment by churik** [comment]
@siphiuel  Stuck on `make watch-desktop` :
```
make watch-desktop
clj -R:dev build.clj watch --platform desktop
make: clj: No such file or directory
make: *** [watch-desktop] Error 1
```
Just in case my `git status`:
```
HEAD detached at origin/fix/5444-desktop-force-node-restart
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   desktop_files/package-lock.json

no changes added to commit (use "git add" and/or "git commit -a")

```

**Comment by vitvly** [comment]
Sorry again. Please run `brew install clojure` - this will install `clj`.

**Comment by churik** [comment]
Can't `react-native run-desktop`:
```
Command failed: ./build.sh -e "node_modules/react-native-i18n/desktop;node_modules/react-native-config/desktop;node_modules/react-native-fs/desktop;node_modules/react-native-http-bridge/desktop;node_modules/react-native-webview-bridge/desktop;node_modules/react-native-keychain/desktop;node_modules/react-native-securerandom/desktop;modules/react-native-status/desktop;modules/react-native-desktop-linking/desktop;modules/react-native-desktop-notification/desktop;node_modules/google-breakpad" -f "../../../../../resources/fonts/SF-Pro-Text-Regular.otf;../../../../../resources/fonts/SF-Pro-Text-Medium.otf;../../../../../resources/fonts/SF-Pro-Text-Light.otf"
CMake Error at CMakeLists.txt:6 (find_package):
  Could not find a package configuration file provided by "ECM" (requested
  version 1.7.0) with any of the following names:

    ECMConfig.cmake
    ecm-config.cmake

  Add the installation prefix of "ECM" to CMAKE_PREFIX_PATH or set "ECM_DIR"
  to a directory containing one of the above files.  If "ECM" provides a
  separate development package or SDK, be sure it has been installed.


make[2]: *** [modules/react-native-desktop-notification/desktop/SnoreNotify_ep-prefix/src/SnoreNotify_ep-stamp/SnoreNotify_ep-configure] Error 1
make[1]: *** [modules/react-native-desktop-notification/desktop/CMakeFiles/SnoreNotify_ep.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by vitvly** [comment]
`brew install kde-mac/kde/kf5-extra-cmake-modules`

**Comment by churik** [comment]
@siphiuel can you please update the doc?

**Pull request merged**